RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitProducerGPU.cc

```diff
@@ -205,46 +215,40 @@ void EcalUncalibRecHitProducerGPU::acquire(edm::Event const& event,
+  auto const& multifitParametersData = setup.getData(multifitParametersToken_);
+
+  auto const& pedestals = setup.getData(pedestalsToken_).getProduct(ctx.stream());
+  auto const& gainRatios = setup.getData(gainRatiosToken_).getProduct(ctx.stream());
+  auto const& pulseShapes = setup.getData(pulseShapesToken_).getProduct(ctx.stream());
+  auto const& pulseCovariances = setup.getData(pulseCovariancesToken_).getProduct(ctx.stream());
+  auto const& samplesCorrelation = setup.getData(samplesCorrelationToken_).getProduct(ctx.stream());
+  auto const& timeBiasCorrections = setup.getData(timeBiasCorrectionsToken_).getProduct(ctx.stream());
+  auto const& timeCalibConstants = timeCalibConstantsData.getProduct(ctx.stream());
+  auto const& multifitParameters = multifitParametersData.getProduct(ctx.stream());
 
   // assign ptrs/values: this is done not to change how things look downstream
-  configParameters_.amplitudeFitParametersEB = multifitParametersProduct.amplitudeFitParametersEB;
-  configParameters_.amplitudeFitParametersEE = multifitParametersProduct.amplitudeFitParametersEE;
-  configParameters_.timeFitParametersEB = multifitParametersProduct.timeFitParametersEB;
-  configParameters_.timeFitParametersEE = multifitParametersProduct.timeFitParametersEE;
-  configParameters_.timeFitParametersSizeEB = multifitParametersHandle_->getValues()[2].get().size();
-  configParameters_.timeFitParametersSizeEE = multifitParametersHandle_->getValues()[3].get().size();
+  configParameters_.amplitudeFitParametersEB = multifitParameters.amplitudeFitParametersEB;
+  configParameters_.amplitudeFitParametersEE = multifitParameters.amplitudeFitParametersEE;
+  configParameters_.timeFitParametersEB = multifitParameters.timeFitParametersEB;
+  configParameters_.timeFitParametersEE = multifitParameters.timeFitParametersEE;
+  configParameters_.timeFitParametersSizeEB = multifitParametersData.getValues()[2].get().size();
+  configParameters_.timeFitParametersSizeEE = multifitParametersData.getValues()[3].get().size();
 
   // bundle up conditions
-  ecal::multifit::ConditionsProducts conditions{pedProduct,
-                                                gainsProduct,
-                                                pulseShapesProduct,
-                                                pulseCovariancesProduct,
-                                                samplesCorrelationProduct,
-                                                timeBiasCorrectionsProduct,
-                                                timeCalibConstantsProduct,
-                                                *sampleMaskHandle_,
-                                                *timeOffsetConstantHandle_,
-                                                timeCalibConstantsHandle_->getOffset(),
-                                                multifitParametersProduct};
+  ecal::multifit::ConditionsProducts conditions{pedestals,
```

Comment:
In principle some of the temporary variables could be dropped along
```suggestion
  ecal::multifit::ConditionsProducts conditions{setup.getData(pedestalsToken_).getProduct(ctx.stream()),
```
but I can also see benefits in the current style.

Reply:
I'm afraid this would become much harder to follow if I put all the `setup.getData(...).getProduct(...)` calls here .

Reply:
Especially since `timeCalibConstantsData` is used in a couple of places, and in some case we call `getProduct(...)`, while in others we don't.